marklogic-data-hub-central/ui/src/util/user-context.tsx

```diff
@@ -40,8 +41,27 @@ const UserProvider: React.FC<{ children: any }> = ({children}) => {
   const [user, setUser] = useState<UserContextInterface>(defaultUserData);
   const sessionUser = localStorage.getItem('dataHubUser');
   const authoritiesService = useContext(AuthoritiesContext);
+  const stompService = useContext(StompContext);
   let sessionCount = 300;
   let sessionTimer = true;
+  let unsubscribeId;
+  const monitorSession = async () => {
+    stompService.onConnected().then(() => {
+      const hubCentralSessionToken = localStorage.getItem('hubCentralSessionToken');
+      stompService.messages.subscribe((message) => {
+        sessionCount = parseInt(JSON.parse(message.body).sessionTimeout);
+      });
+      stompService.subscribe(`/topic/sessionStatus/${hubCentralSessionToken}`, (msgId: string) => {
+        unsubscribeId = msgId;
+      });
+    });
+  };
```

Comment:
Do we need some sort of cross tab communication to propagate the timeout events on all tabs  https://dev.to/naismith/cross-tab-communication-with-javascript-1hc9

Reply:
Each tab will be subscribed to the same sessionToken via WebSockets, so each tab will have sessionCount updated.